.pre-commit-config.yaml

```diff
@@ -0,0 +1,26 @@
+repos:
+  - repo: https://github.com/psf/black
+    rev: 22.6.0
+    hooks:
+      - id: black
+        files: \.py$|\.pyi$
+        language_version: python3.8
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.7.1
+    hooks:
+      - id: prettier
+        files: \.js$|\.jsx$|\.ts$|\.tsx$
+  - repo: local
+    hooks:
+      - id: run-typecheck
+        name: Run typecheck
+        entry: yarn --cwd frontend tsc --noEmit
+        files: \.js$|\.jsx$|\.ts$|\.tsx$
+        language: node
+        pass_filenames: false
+      - id: run-eslint
```

Comment:
What happens if the linter detects some problems? Is the commit aborted? Is there an error message that is shown to the user even when git commit is not triggered from the terminal?

Reply:
@LukasMasuch thanks for your question, it reminded me that I should add some explanation both in the code and [Contributing page](https://github.com/streamlit/streamlit/wiki/Contributing). I've added **Pre-commit hooks** section into [Contributing page](https://github.com/streamlit/streamlit/wiki/Contributing) and some comments into `.pre-commit-config.yaml`.

Here is my comment:
```
# Pre-commit configuration file,
# when Streamlit's pre-commit detects that one of the linters has failed,
# it automatically lints the files and does not allow the commit to pass. 
# Please review the changes after lint has failed and commit them again,
# the second commit should pass,
# because the files were linted after trying to do the first commit.
```